transport/src/main/java/io/el/channel/AbstractChannelHandlerContext.java

```diff
@@ -111,16 +111,74 @@ private AbstractChannelHandlerContext findContextInbound() {
     return ctx;
   }
 
+  private AbstractChannelHandlerContext findContextOutbound() {
+    AbstractChannelHandlerContext ctx = this;
+    do {
+      ctx = ctx.prev;
+    } while (ctx.executionFlag != FLAG_OUTBOUND);
+    return ctx;
+  }
+
+  public ChannelPromise newPromise() {
+    return new DefaultChannelPromise(channel(), eventLoop());
+  }
+
   @Override
   public ChannelPromise bind(SocketAddress localAddress) {
-    // TODO: implement me
-    return null;
+    return this.bind(localAddress, newPromise());
   }
 
   @Override
   public ChannelPromise bind(SocketAddress localAddress, ChannelPromise promise) {
-    // TODO: implement me
-    return null;
+    ObjectUtil.checkNotNull(localAddress, "localAddress");
+    if (isNotValidPromise(promise)) {
+      // canceled
+      return promise;
+    }
+
+    final AbstractChannelHandlerContext next = findContextOutbound();
+    final EventLoop eventLoop = next.eventLoop();
+    if (eventLoop.inEventLoop()) {
+      next.bind(localAddress, promise);
```

Comment:
Shouldn't we call the handler of context? It looks like infinite recursive call

Reply:
Oh, I missed it. I'll fix it.

Reply:
@zeroFruit I fixed it! https://github.com/zeroFruit/el/pull/54/commits/ed857971c36025be03f78adc19bf04d9a9d020c9